integration/container/attach_test.go

```diff
@@ -0,0 +1,206 @@
+// +build !windows
+
+package container
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/internal/container"
+	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
+	"gotest.tools/poll"
+	"gotest.tools/skip"
+)
+
+// #9860 Make sure attach ends when container ends (with no errors)
+func TestAttachClosedOnContainerStop(t *testing.T) {
+	skip.If(t, testEnv.IsRemoteDaemon, "test requires daemon on the same host")
+
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := testEnv.APIClient()
+
+	cID := container.Run(t, ctx, client, container.WithTty(true),
+		container.WithCmd("/bin/sh", "-c", `trap 'exit 0' SIGTERM; while true; do sleep 1; done`),
+		func(c *container.TestContainerConfig) {
+			c.Config.OpenStdin = true
+		},
+	)
+	defer client.ContainerRemove(context.Background(), cID, types.ContainerRemoveOptions{Force: true})
+	poll.WaitOn(t, container.IsInState(ctx, client, cID, "running"), poll.WithDelay(5*time.Second))
```

Comment:
Looks like the original version used `waitRun`, which has a _timeout_ of 5 seconds; https://github.com/moby/moby/blob/89ce20fa250070b02a9f05ef6d5dc9270f1181ad/integration-cli/docker_utils_test.go#L319-L331


However here it changed to a `delay` of 5 seconds (and the default _timeout_ of 10 seconds).

For this, I think we can keep the default 100ms interval (perhaps even keep the default 10sec timeout, but we could make that shorter, e.g. `poll.WithTimeout(3*time.Second)`)

Reply:
Thank you for the review, updated the poll to use 100ms interval (keeping the default timeout).